pkg/river/encoding/json.go

```diff
@@ -0,0 +1,149 @@
+package encoding
```

Comment:
nit: I think this file should be called encoding.go so it's clearer that it's the entrypoint of the package.